tests/Functional/ORMModelFactoryTest.php

```diff
@@ -493,6 +495,24 @@ public function embeddables_are_never_persisted(): void
         $this->assertSame('another address', $object2->getValue());
     }
 
+    /**
+     * @test
+     */
+    public function can_create_factories_for_rich_domain_models(): void
+    {
+        $post = RichPostFactory::createOne([
+            'comments' => RichCommentFactory::new()->many(5),
+        ]);
+
+        RichPostFactory::assert()->count(1);
+        RichCommentFactory::assert()->count(5);
+        $this->assertCount(5, $post->getComments());
+
+        foreach (RichCommentFactory::all() as $comment) {
+            $this->assertSame($post->getId(), $comment->getPost()->getId());
```

Comment:
I don't understand how this works :upside_down_face: 
we never say "the post we're creating is the one to use for all comments". How is it possible that a new post is not created for each comment?

Reply:
There is some magic in `Factory` that handles this when creating relationships.

Reply:
nice! it was this particular magic I didn't know about when I wrote the issue